### ISSUE TITLE
Checking P's weights

### ISSUE THREAD

**Issue by mayanks43** [DateDocta/DatingGame#3]
According to the architecture, "In each response, P may modify 5% of the criteria (P may choose which ones) by 20% each with respect to the original weights. P must ensure that the sum of P's positive weights must be 1 and the sum of the negative weights must be -1.". 

Is the architecture doing anything to confirm this is the case (shouldn't allow more than 5% attributes and shouldn't allow more than 20% change for each) (couldn't find it in game.coffee)? I remember Prof. saying that the architecture should use the old weights in case P fails to satisfy these constraints.

**Comment by DateDocta** [comment]
Yes, check the matchmaker.js/coffee and the player.coffee.  

These functions here.
[matchmaker](https://github.com/DateDocta/DatingGame/blob/master/coffeescripts/matchmaker.coffee#L16)

and

[player](https://github.com/DateDocta/DatingGame/blob/master/coffeescripts/player.coffee#L63)

**Important things to note:**
If you send something invalid, I will log it on the server, but I do not want this to break the game.  This is what I have done.

On both the matchmaker and player, if we are not dealing with the first candidate and we receive invalid weights from someone, we will not use the invalid weights and instead take the last acceptable valid candidate.

On the first turn. If player sends something invalid, means something out of the allowable threshold, then we have a big message print out and we will reject the game.  (It should be easy to send an allowable candidate on the first attempt)

On the first turn of the matchmaker, if you send an invalid candidate, we will make it valid for you with simple values and continue.  You will know this is happening when you receive weights that weren't yours as a response to one of your candidates.  

**TLDR** Make sure to send valid candidates.  If you don't we will just take your last valid candidate.  If you don't have any valid candidates, the player game will be considered invalid, and we will make a candidate for you for the matchmaker.  The candidate we make for you with the matchmaker will not be a good one, so just send us good ones :)

**Comment by mayanks43** [comment]
Awesome. sounds good. :+1:
